content/articles/2018-12-textrecipes-0-0-1.Rmarkdown

```diff
@@ -0,0 +1,84 @@
+  author: Roman Kraft
+---
+
+```{r setup, include = FALSE}
+library(textrecipes)
+knitr::opts_chunk$set(
+  collapse = TRUE, comment = "#>", 
+  fig.width = 7, 
+  fig.align = 'center',
+  fig.asp = 0.618, # 1 / phi
+  out.width = "700px"
+)
+```
+
+We're delighted to announce the release of [textrecipes 0.0.1](https://github.com/tidymodels/textrecipes) on CRAN. [`textrecipes`](https://github.com/tidymodels/textrecipes) implements a collection of new steps for the [`recipes`](https://github.com/tidymodels/recipes) package to deal with text preprocessing. [`textrecipes`](https://github.com/tidymodels/textrecipes) is still in early development so any and all feedback is highly appreciated.
+
+You can install it by running:
+
+```{r, eval = FALSE}
+install.packages("textrecipes")
+```
+
+## New steps
+
+The steps introduces here can be split into 3 types. The steps
+
+1. converts from characters to list-columns and vice versa
+1. modifies the elements in list-columns
+1. converts list-columns to numerics
+
+this allows for great flexibility in the preprocessing tasks that can be done while staying inside the [`recipes`](https://github.com/tidymodels/recipes) framework. This also prevents having one single step having a million arguments.
+
+## Workflows
+
+First we start by creating a [`recipe`](https://tidymodels.github.io/recipes/reference/recipe.html) object from the original data.
+
+```{r}
+data("okc_text")
+rec_obj <- recipe(~ ., okc_text)
+
+rec_obj
+```
+
+The workflow in `textrecipes` so far starts with [`step_tokenize()`](https://tidymodels.github.io/textrecipes/reference/step_tokenize.html) followed by a combination of type 1 and 2 steps ending with a type 3 step. [`step_tokenize()`](https://tidymodels.github.io/textrecipes/reference/step_tokenize.html) wraps the [`tokenizers`](https://github.com/ropensci/tokenizers) package for tokenization, but other tokenization functions can be utilized using the `custom_token` argument, more information concerning arguments can be found in the documentation. The shortest possible recipes are [`step_tokenize()`](https://tidymodels.github.io/textrecipes/reference/step_tokenize.html) directly followed by a type 3 step.
+
+```{r}
+### Feature hashing done on word tokens
+rec_obj %>%
+  step_tokenize(essay0) %>% # token argument defaults to "words"
+  step_texthash(essay0)
+
+### Counting chacter occurrences
+rec_obj %>%
+  step_tokenize(essay0, token = "character") %>%
+  step_tf(essay0)
+```
+
+If one wanted to calculate the word count of the top 100 most frequently used words after stemming is performed, type 2 steps are needed. Here we use [`step_stem()`](https://tidymodels.github.io/textrecipes/reference/step_stem.html) to perform stemming using the [`SnowballC`](https://CRAN.R-project.org/package=SnowballC) package and [`step_tokenfilter()`](https://tidymodels.github.io/textrecipes/reference/step_tokenfilter.html) to only keep the most 100 frequent tokens.
```

Comment:
* type-2 steps
* Remove backticks from SnowballC
* Opinions vary on split infinitives, but I'd go with "to keep only…"

Reply:
> …the most 100 frequent tokens.

I think you mean the 100 most frequent.